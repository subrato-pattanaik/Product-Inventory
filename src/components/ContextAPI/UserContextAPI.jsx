import { createContext } from "react";
export const UserContextAPI = createContext(null);
