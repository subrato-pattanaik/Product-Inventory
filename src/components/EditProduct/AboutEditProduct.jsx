import React from "react";
import { Container } from "react-bootstrap";

function AboutEditProduct() {
  return (
    <Container className="pt-3">
      <div className="text-center">
        <h4 className="mb-n2 display-6">About Product List</h4>
        <hr className="mb-4 bg-warning" />
        <div className="small lead mb-4">
          <p>Here you can edit the products in the inventory</p>
          <p>You can view all the products in the product List</p>
        </div>
      </div>
    </Container>
  );
}

export default AboutEditProduct;
