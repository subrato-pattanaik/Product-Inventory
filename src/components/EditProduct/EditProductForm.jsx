import React, { useState, useEffect } from "react";
import { withRouter, useHistory, useLocation, Prompt } from "react-router-dom";
import { useForm } from "react-hook-form";

import { Col, Button, Form } from "react-bootstrap";

import axios from "axios";

function EditProductForm() {
  let location = useLocation();
  const { pid } = location.state;
  const [sureEdit, setSureEdit] = useState(false);
  const [validated, setValidated] = useState(false);
  const [isDirty, setIsDirty] = useState(false);
  const [editUser, setEditUser] = useState({
    id: 0,
    productName: "",
    productDesc: "",
    manufacturer: "",
    price: "",
    quantiy: "",
    select: false,
  });

  const { register, handleSubmit, reset } = useForm({
    defaultValues: editUser,
  });

  useEffect(() => {
    async function fetchData() {
      await axios.get(`http://localhost:4000/products/${pid}`).then((res) => {
        setEditUser(res.data);
        reset(res.data);
      });
    }
    fetchData();
  }, [pid, reset]);

  let navigation = useHistory();
  const onSubmit = (formData) => {
    formData.select = false;
    axios.put(`http://localhost:4000/products/${pid}`, formData);
    setSureEdit(true);
    setIsDirty(false);
    navigation.push({
      pathname: "/productList",
      state: { added: "pass" },
    });

    setValidated(true);
  };

  return (
    <>
      <Form noValidate validated={validated} onSubmit={handleSubmit(onSubmit)}>
        <Form.Row>
          <Form.Group as={Col} className="mr-5">
            <Form.Label column>Product Name</Form.Label>
            <Form.Control
              name="productName"
              placeholder="Enter product Name"
              required
              onChange={() => setIsDirty(true)}
              ref={register}
            />
            <Form.Control.Feedback type="invalid">
              Please enter valid product name
            </Form.Control.Feedback>
          </Form.Group>

          <Form.Group as={Col}>
            <Form.Label column>Product Description</Form.Label>
            <Form.Control
              name="productDesc"
              placeholder="Enter product description"
              required
              onChange={() => setIsDirty(true)}
              ref={register}
            />
            <Form.Control.Feedback type="invalid">
              Please enter valid product description
            </Form.Control.Feedback>
          </Form.Group>
        </Form.Row>

        <Form.Row>
          <Form.Group as={Col} className="mr-5">
            <Form.Label column>Manufacturer</Form.Label>
            <Form.Control
              name="manufacturer"
              placeholder="Enter Manufacturer"
              required
              onChange={() => setIsDirty(true)}
              ref={register}
            />
            <Form.Control.Feedback type="invalid">
              Please enter valid manufacturer
            </Form.Control.Feedback>
          </Form.Group>

          <Form.Group as={Col} controlId="formBasicPrice">
            <Form.Label column>Price</Form.Label>
            <Form.Control
              type="number"
              name="price"
              placeholder="Enter Price"
              required
              onChange={() => setIsDirty(true)}
              ref={register}
            />
            <Form.Control.Feedback type="invalid">
              Please enter valid Price
            </Form.Control.Feedback>
          </Form.Group>
        </Form.Row>

        <Form.Row>
          <Form.Group as={Col}>
            <Form.Label column>Quantity</Form.Label>
            <Form.Control
              type="number"
              name="quantity"
              placeholder="Enter Quantity"
              required
              onChange={() => setIsDirty(true)}
              ref={register}
              max="100"
            />
            <Form.Control.Feedback type="invalid">
              Please Quantity in number format and quanitity should be less than
              100
            </Form.Control.Feedback>
          </Form.Group>
        </Form.Row>
        <div className="d-flex justify-content-end">
          <Button variant="warning" type="submit" className="text-white mt-5">
            Edit Product
          </Button>
          <Button
            variant="danger"
            onClick={() => navigation.push("./productList")}
            className=" mt-5 ml-3"
          >
            Cancel
          </Button>
        </div>
      </Form>
      <Prompt
        when={isDirty}
        message="Are you sure you want to leave this edit product page?"
      />
      <Prompt when={sureEdit} message="Are you sure you want to edit?" />
    </>
  );
}

export default withRouter(EditProductForm);
