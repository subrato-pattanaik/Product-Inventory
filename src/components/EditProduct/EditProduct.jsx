import React, { useContext } from "react";
import { Col, Container, Row } from "react-bootstrap";
import { UserContextAPI } from "../ContextAPI/UserContextAPI";
import EditProductForm from "./EditProductForm";
import { useHistory, withRouter } from "react-router-dom";
import AboutEditProduct from "./AboutEditProduct";
function EditProduct() {
  const { user } = useContext(UserContextAPI);
  let navigation = useHistory();

  return (
    <>
      {!user ? (
        <>{navigation.push("/Login")}</>
      ) : (
        <Container fluid className="mt-4 mb-5">
          <Row>
            <Col sm={2} className="border-right bg-light">
              <AboutEditProduct />
            </Col>
            <Col sm={10}>
              <Container>
                <Row className="mt-2">
                  <Col sm={4}>
                    <h3 className="display-5 ml-4">Edit Product</h3>
                  </Col>
                </Row>
                <hr />
                <Row>
                  <Col>
                    <EditProductForm />
                  </Col>
                </Row>
              </Container>
            </Col>
          </Row>
        </Container>
      )}
    </>
  );
}

export default withRouter(EditProduct);
