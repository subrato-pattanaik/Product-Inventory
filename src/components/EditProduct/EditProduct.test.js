import React from "react";
import { Router } from "react-router-dom";
import EditProduct from "./EditProduct";
import { mount } from "enzyme";
import "mutationobserver-shim";
import { UserContextAPI } from "../ContextAPI/UserContextAPI";
//import { Form } from "react-bootstrap";
import { createBrowserHistory } from "history";
describe("Edit Page Component", () => {
  let wrapper;

  beforeEach(() => {
    const history = createBrowserHistory();
    const state = { pid: 1 };
    history.push("/EditProduct", state);
    wrapper = mount(
      <Router history={history}>
        <UserContextAPI.Provider value={{ user: 1 }}>
          <EditProduct.WrappedComponent />
        </UserContextAPI.Provider>
      </Router>
    );
  });

  it("should render About Edit Product component", () => {
    expect(wrapper.find("AboutEditProduct").exists()).toEqual(true);
  });

  it("should render  Edit Product form component", () => {
    expect(wrapper.find("EditProductForm")).toHaveLength(1);
  });
});
