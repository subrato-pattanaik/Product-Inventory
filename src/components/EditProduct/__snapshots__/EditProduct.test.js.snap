// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Page Component should render Edit Product component 1`] = `
<Router
  history={
    Object {
      "action": "PUSH",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 2,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "6ld7mx",
        "pathname": "/EditProduct",
        "search": "",
        "state": Object {
          "pid": 1,
        },
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <EditProduct>
    <Container
      className="mt-4 mb-5"
      fluid={true}
    >
      <div
        className="mt-4 mb-5 container-fluid"
      >
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Col
              className="border-right bg-light"
              sm={2}
            >
              <div
                className="border-right bg-light col-sm-2"
              >
                <AboutEditProduct>
                  <Container
                    className="pt-3"
                    fluid={false}
                  >
                    <div
                      className="pt-3 container"
                    >
                      <div
                        className="text-center"
                      >
                        <h4
                          className="mb-n2 display-6"
                        >
                          About Product List
                        </h4>
                        <hr
                          className="mb-4 bg-warning"
                        />
                        <div
                          className="small lead mb-4"
                        >
                          <p>
                            Here you can edit the products in the inventory
                          </p>
                          <p>
                            You can view all the products in the product List
                          </p>
                        </div>
                      </div>
                    </div>
                  </Container>
                </AboutEditProduct>
              </div>
            </Col>
            <Col
              sm={10}
            >
              <div
                className="col-sm-10"
              >
                <Container
                  fluid={false}
                >
                  <div
                    className="container"
                  >
                    <Row
                      className="mt-2"
                      noGutters={false}
                    >
                      <div
                        className="mt-2 row"
                      >
                        <Col
                          sm={4}
                        >
                          <div
                            className="col-sm-4"
                          >
                            <h3
                              className="display-5 ml-4"
                            >
                              Edit Product
                            </h3>
                          </div>
                        </Col>
                      </div>
                    </Row>
                    <hr />
                    <Row
                      noGutters={false}
                    >
                      <div
                        className="row"
                      >
                        <Col>
                          <div
                            className="col"
                          >
                            <withRouter(EditProductForm)>
                              <EditProductForm
                                history={
                                  Object {
                                    "action": "PUSH",
                                    "block": [Function],
                                    "createHref": [Function],
                                    "go": [Function],
                                    "goBack": [Function],
                                    "goForward": [Function],
                                    "length": 2,
                                    "listen": [Function],
                                    "location": Object {
                                      "hash": "",
                                      "key": "6ld7mx",
                                      "pathname": "/EditProduct",
                                      "search": "",
                                      "state": Object {
                                        "pid": 1,
                                      },
                                    },
                                    "push": [Function],
                                    "replace": [Function],
                                  }
                                }
                                location={
                                  Object {
                                    "hash": "",
                                    "key": "6ld7mx",
                                    "pathname": "/EditProduct",
                                    "search": "",
                                    "state": Object {
                                      "pid": 1,
                                    },
                                  }
                                }
                                match={
                                  Object {
                                    "isExact": false,
                                    "params": Object {},
                                    "path": "/",
                                    "url": "/",
                                  }
                                }
                              >
                                <Form
                                  inline={false}
                                  noValidate={true}
                                  onSubmit={[Function]}
                                  validated={false}
                                >
                                  <form
                                    className=""
                                    noValidate={true}
                                    onSubmit={[Function]}
                                  >
                                    <FormRow>
                                      <div
                                        className="form-row"
                                      >
                                        <FormGroup
                                          as={
                                            Object {
                                              "$$typeof": Symbol(react.forward_ref),
                                              "displayName": "Col",
                                              "render": [Function],
                                            }
                                          }
                                          className="mr-5"
                                        >
                                          <Col
                                            className="mr-5 form-group"
                                          >
                                            <div
                                              className="mr-5 form-group col"
                                            >
                                              <FormLabel
                                                column={true}
                                                srOnly={false}
                                              >
                                                <Col
                                                  as="label"
                                                  className="form-label col-form-label"
                                                >
                                                  <label
                                                    className="form-label col-form-label col"
                                                  >
                                                    Product Name
                                                  </label>
                                                </Col>
                                              </FormLabel>
                                              <FormControl
                                                name="productName"
                                                onChange={[Function]}
                                                placeholder="Enter product Name"
                                                required={true}
                                              >
                                                <input
                                                  className="form-control"
                                                  name="productName"
                                                  onChange={[Function]}
                                                  placeholder="Enter product Name"
                                                  required={true}
                                                />
                                              </FormControl>
                                              <Feedback
                                                type="invalid"
                                              >
                                                <div
                                                  className="invalid-feedback"
                                                >
                                                  Please enter valid product name
                                                </div>
                                              </Feedback>
                                            </div>
                                          </Col>
                                        </FormGroup>
                                        <FormGroup
                                          as={
                                            Object {
                                              "$$typeof": Symbol(react.forward_ref),
                                              "displayName": "Col",
                                              "render": [Function],
                                            }
                                          }
                                        >
                                          <Col
                                            className="form-group"
                                          >
                                            <div
                                              className="form-group col"
                                            >
                                              <FormLabel
                                                column={true}
                                                srOnly={false}
                                              >
                                                <Col
                                                  as="label"
                                                  className="form-label col-form-label"
                                                >
                                                  <label
                                                    className="form-label col-form-label col"
                                                  >
                                                    Product Description
                                                  </label>
                                                </Col>
                                              </FormLabel>
                                              <FormControl
                                                name="productDesc"
                                                onChange={[Function]}
                                                placeholder="Enter product description"
                                                required={true}
                                              >
                                                <input
                                                  className="form-control"
                                                  name="productDesc"
                                                  onChange={[Function]}
                                                  placeholder="Enter product description"
                                                  required={true}
                                                />
                                              </FormControl>
                                              <Feedback
                                                type="invalid"
                                              >
                                                <div
                                                  className="invalid-feedback"
                                                >
                                                  Please enter valid product description
                                                </div>
                                              </Feedback>
                                            </div>
                                          </Col>
                                        </FormGroup>
                                      </div>
                                    </FormRow>
                                    <FormRow>
                                      <div
                                        className="form-row"
                                      >
                                        <FormGroup
                                          as={
                                            Object {
                                              "$$typeof": Symbol(react.forward_ref),
                                              "displayName": "Col",
                                              "render": [Function],
                                            }
                                          }
                                          className="mr-5"
                                        >
                                          <Col
                                            className="mr-5 form-group"
                                          >
                                            <div
                                              className="mr-5 form-group col"
                                            >
                                              <FormLabel
                                                column={true}
                                                srOnly={false}
                                              >
                                                <Col
                                                  as="label"
                                                  className="form-label col-form-label"
                                                >
                                                  <label
                                                    className="form-label col-form-label col"
                                                  >
                                                    Manufacturer
                                                  </label>
                                                </Col>
                                              </FormLabel>
                                              <FormControl
                                                name="manufacturer"
                                                onChange={[Function]}
                                                placeholder="Enter Manufacturer"
                                                required={true}
                                              >
                                                <input
                                                  className="form-control"
                                                  name="manufacturer"
                                                  onChange={[Function]}
                                                  placeholder="Enter Manufacturer"
                                                  required={true}
                                                />
                                              </FormControl>
                                              <Feedback
                                                type="invalid"
                                              >
                                                <div
                                                  className="invalid-feedback"
                                                >
                                                  Please enter valid manufacturer
                                                </div>
                                              </Feedback>
                                            </div>
                                          </Col>
                                        </FormGroup>
                                        <FormGroup
                                          as={
                                            Object {
                                              "$$typeof": Symbol(react.forward_ref),
                                              "displayName": "Col",
                                              "render": [Function],
                                            }
                                          }
                                          controlId="formBasicPrice"
                                        >
                                          <Col
                                            className="form-group"
                                          >
                                            <div
                                              className="form-group col"
                                            >
                                              <FormLabel
                                                column={true}
                                                srOnly={false}
                                              >
                                                <Col
                                                  as="label"
                                                  className="form-label col-form-label"
                                                  htmlFor="formBasicPrice"
                                                >
                                                  <label
                                                    className="form-label col-form-label col"
                                                    htmlFor="formBasicPrice"
                                                  >
                                                    Price
                                                  </label>
                                                </Col>
                                              </FormLabel>
                                              <FormControl
                                                name="price"
                                                onChange={[Function]}
                                                placeholder="Enter Price"
                                                required={true}
                                                type="number"
                                              >
                                                <input
                                                  className="form-control"
                                                  id="formBasicPrice"
                                                  name="price"
                                                  onChange={[Function]}
                                                  placeholder="Enter Price"
                                                  required={true}
                                                  type="number"
                                                />
                                              </FormControl>
                                              <Feedback
                                                type="invalid"
                                              >
                                                <div
                                                  className="invalid-feedback"
                                                >
                                                  Please enter valid Price
                                                </div>
                                              </Feedback>
                                            </div>
                                          </Col>
                                        </FormGroup>
                                      </div>
                                    </FormRow>
                                    <FormRow>
                                      <div
                                        className="form-row"
                                      >
                                        <FormGroup
                                          as={
                                            Object {
                                              "$$typeof": Symbol(react.forward_ref),
                                              "displayName": "Col",
                                              "render": [Function],
                                            }
                                          }
                                        >
                                          <Col
                                            className="form-group"
                                          >
                                            <div
                                              className="form-group col"
                                            >
                                              <FormLabel
                                                column={true}
                                                srOnly={false}
                                              >
                                                <Col
                                                  as="label"
                                                  className="form-label col-form-label"
                                                >
                                                  <label
                                                    className="form-label col-form-label col"
                                                  >
                                                    Quantity
                                                  </label>
                                                </Col>
                                              </FormLabel>
                                              <FormControl
                                                max="100"
                                                name="quantity"
                                                onChange={[Function]}
                                                placeholder="Enter Quantity"
                                                required={true}
                                                type="number"
                                              >
                                                <input
                                                  className="form-control"
                                                  max="100"
                                                  name="quantity"
                                                  onChange={[Function]}
                                                  placeholder="Enter Quantity"
                                                  required={true}
                                                  type="number"
                                                />
                                              </FormControl>
                                              <Feedback
                                                type="invalid"
                                              >
                                                <div
                                                  className="invalid-feedback"
                                                >
                                                  Please Quantity in number format and quanitity should be less than 100
                                                </div>
                                              </Feedback>
                                            </div>
                                          </Col>
                                        </FormGroup>
                                      </div>
                                    </FormRow>
                                    <div
                                      className="d-flex justify-content-end"
                                    >
                                      <Button
                                        active={false}
                                        className="text-white mt-5"
                                        disabled={false}
                                        type="submit"
                                        variant="warning"
                                      >
                                        <button
                                          className="text-white mt-5 btn btn-warning"
                                          disabled={false}
                                          type="submit"
                                        >
                                          Edit Product
                                        </button>
                                      </Button>
                                      <Button
                                        active={false}
                                        className=" mt-5 ml-3"
                                        disabled={false}
                                        onClick={[Function]}
                                        variant="danger"
                                      >
                                        <button
                                          className=" mt-5 ml-3 btn btn-danger"
                                          disabled={false}
                                          onClick={[Function]}
                                          type="button"
                                        >
                                          Cancel
                                        </button>
                                      </Button>
                                    </div>
                                  </form>
                                </Form>
                                <Prompt
                                  message="Are you sure you want to leave this edit product page?"
                                  when={false}
                                />
                                <Prompt
                                  message="Are you sure you want to edit?"
                                  when={false}
                                />
                              </EditProductForm>
                            </withRouter(EditProductForm)>
                          </div>
                        </Col>
                      </div>
                    </Row>
                  </div>
                </Container>
              </div>
            </Col>
          </div>
        </Row>
      </div>
    </Container>
  </EditProduct>
</Router>
`;
